examples/src/index.js

```diff
@@ -44,6 +44,16 @@ Or:
 
 This is more text.
 
+# Horizontal Rules
+Hyphens:
+---
+
+Asterisks:
+***
+
+Underscores:
+___
+
```

Comment:
Could it be that they need a space above them? 

```
Hyphens:

---

Asterisks:
...
```

Reply:
You're right. The hyphens under a text creates a heading. 
https://www.markdownguide.org/basic-syntax/#alternate-syntax

The reason, why the horizontal rules are not parsed properly is `slateTransformer.fromMarkdown(defaultMarkdown)`.
They are typed as `paragraph` instead of `horizontal_rule`.

See:
1. https://github.com/accordproject/slate-markdown-transform/blob/master/src/slateToCommonMarkAst.js#L64-L65
2. https://github.com/accordproject/slate-markdown-transform/blob/master/src/ToSlateVisitor.js#L193-L199

So from rich text to markdown works fine.

Reply:
> You're right. The hyphens under a text creates a heading.
> https://www.markdownguide.org/basic-syntax/#alternate-syntax
> 
> The reason, why the horizontal rules are not parsed properly is `slateTransformer.fromMarkdown(defaultMarkdown)`.
> They are typed as `paragraph` instead of `horizontal_rule`.
> 
> See:
> 
> 1. https://github.com/accordproject/slate-markdown-transform/blob/master/src/slateToCommonMarkAst.js#L64-L65
> 2. https://github.com/accordproject/slate-markdown-transform/blob/master/src/ToSlateVisitor.js#L193-L199
> 
> So from rich text to markdown works fine.

Those are from an old project (see in the project title). The current markdown to slate transform is in: https://github.com/accordproject/markdown-transform

There seems to be a bug indeed in the `markdown-transform` when it comes to thematic breaks, but I'm not sure what the expected Slate DOM should be. I'm opening an issue about this so it can be discussed there.

Reply:
> You're right. The hyphens under a text creates a heading.
> https://www.markdownguide.org/basic-syntax/#alternate-syntax
> 
> The reason, why the horizontal rules are not parsed properly is `slateTransformer.fromMarkdown(defaultMarkdown)`.
> They are typed as `paragraph` instead of `horizontal_rule`.
> 
> See:
> 
> 1. https://github.com/accordproject/slate-markdown-transform/blob/master/src/slateToCommonMarkAst.js#L64-L65
> 2. https://github.com/accordproject/slate-markdown-transform/blob/master/src/ToSlateVisitor.js#L193-L199
> 
> So from rich text to markdown works fine.

Those are from an old project (see in the project title). The current markdown to slate transform is in: https://github.com/accordproject/markdown-transform



> @waahhhh can you try what I suggested here? [#141 (comment)](https://github.com/accordproject/markdown-editor/pull/141#discussion_r338195199)

Follow up, I think the `markdown-transform` gobbles thematic breaks, so we probably need a fix there first. See: https://github.com/accordproject/markdown-transform/issues/99